### PR TITLE
dex: support hive cluster

### PR DESCRIPTION
**Pull request by hongkailiu**
https://issues.redhat.com/browse/DPTP-2187

/hold
wait for https://github.com/openshift/release/pull/18685

/cc @alvaroaleman @jupierce

### CODE REVIEW

**Review by stevekuznetsov — APPROVED** [review]

File: clusters/app.ci/dex/manifests.yaml [hunk 1]
```diff
@@ -167,6 +172,16 @@ spec:
             secretKeyRef:
               name: dpcr-secret
               key: dpcr-id
+        - name: hive-SECRET
+          valueFrom:
+            secretKeyRef:
+              name: hive-secret
+              key: hive-secret
+        - name: hive-ID
```
Comment (alvaroaleman) [inline clusters/app.ci/dex/manifests.yaml:180]: This doesn't match the above env var

File: clusters/app.ci/dex/manifests.yaml [hunk 2]
```diff
@@ -167,6 +172,16 @@ spec:
             secretKeyRef:
               name: dpcr-secret
               key: dpcr-id
+        - name: hive-SECRET
```
Comment (alvaroaleman) [inline clusters/app.ci/dex/manifests.yaml:175]: This doesn't match the above env var either

/lgtm cancel

File: clusters/app.ci/dex/manifests.yaml (hunk 1 above)
Comment (hongkailiu) [reply clusters/app.ci/dex/manifests.yaml]: Updated.

File: clusters/app.ci/dex/manifests.yaml (hunk 2 above)
Comment (hongkailiu) [reply clusters/app.ci/dex/manifests.yaml]: updated.

**Comment by hongkailiu** [comment]
```console
$ oc get secret -n dex hive-secret
NAME          TYPE     DATA   AGE
hive-secret   Opaque   2      8m23s
```

/hold cancel

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**